leapp/utils/report.py

```diff
@@ -139,4 +139,13 @@ def generate_report_file(messages_to_report, context, path, report_schema='1.1.0
                     groups = msg.pop('groups', [])
                     msg['flags'] = [g for g in groups if g in _DEPRECATION_FLAGS]
                     msg['tags'] = [g for g in groups if g not in _DEPRECATION_FLAGS]
+            if report_schema_tuple == (1, 2, 0):
+                # DEPRECATED: flags, tags
+                # NOTE(pstodulk): This is a temporary solution to ensure that
+                # flags are not present in the JSON output as we need to re-introduce
+                # flags internally to ensure that any inhibitor checks will not
+                # be negatively affected until we drop flags and tags completely.
+                messages_to_report = list(messages_to_report)
```

Comment:
Maybe I am too tired, but why is the conversion to list necessary?

We are iterating over it on the next line, however, we know that the collection is iterable, otherwise the conversion to list would fail. And we are not modifying the collection itself as we iterate, but rather the individual `msg` dictionaries, therefore, we should be OK without creating a list from the messages.

(Tested using the following snippet)
```python3
a_list = [{'flags': ['someflags']}]
for msg in a_list:
    msg.pop('flags')
```

Reply:
It is not strictly necessary, the reason is mentioned here:
https://github.com/oamg/leapp/blob/567400d46a844d744b62fb938be81fe50ab9e342/leapp/utils/report.py#L131

Basically I am just keeping consistency with the rest of the code in the function.